### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by grulicht** [gotenberg/gotenberg#583] [Q&A]

**Comment by gulien** [comment]
Hello @grulicht,

You want to add basic auth to your Gotenberg instance?

**Reply by grulicht** [reply]
Yes, on the my web.

**Reply by gulien** [reply]
I don't recommend exposing your instances to the external world.

Your best option is to put a public API in front of your Gotenberg instance that handles:

- Access control
- Rate limiting
- etc.

**Reply by grulicht** [reply]
Okey, thanks and how put a public API in front my Gotenberg?
Can I set specific IP addresses in the docker container from which I can call, for example, your demo and thus prevent access to the world?

**Reply by grulicht** [reply]
With basic auth on my web I think this:
I would specify in docker-compose.yml:
-e GOTENBERG_BASIC_AUTH_USERNAME=name \
-e GOTENBERG_BASIC_AUTH_PASSWORD=pass \

and after your demo will look like:
curl \ 
--request POST 'https://my.domain/forms/chromium/convert/url' \
--form 'url="https://sparksuite.github.io/simple-html-invoice-template/"' \
-o my.pdf \
-u name:pass

**Could it be like this?**

**Reply by gulien** [reply]
There are currently no option for adding basic auth in Gotenberg. You may create your own Gotenberg container using the https://github.com/gotenberg/gotenberg-template-repository and coding a module that adds a middleware for basic auth. But it's easier to have a proxy like Traefik or Nginx in front of your container that handles basic auth for you.

**Reply by gulien** [reply]
I don't understand what's you're expecting from me here 😅
